### ISSUE TITLE
Improve errors when incorrect geometry column is passed in CQL_QUERY

### ISSUE THREAD

**Issue by krishnaglodha** [geopython/pygeoapi#1081]
**Description**
I'm running query as 
`http://localhost:4501/collections/Popular%20places%20from%20database/items?f=json&filter=WITHIN(geom, POLYGON ((70.12683873493475 20.939807391818093, 70.12683873493475 16.187786570989473, 76.0788015865644 16.187786570989473, 76.0788015865644 20.939807391818093, 70.12683873493475 20.939807391818093)))`

It works fine, but when I pass incorrect geometry names, it is not giving standard error, instead showing this
<img width="1440" alt="Screenshot 2022-12-26 at 10 42 32 PM" src="https://user-images.githubusercontent.com/47075664/209571157-beb1e5ca-610f-4805-b5b3-cb0263bef330.png">

**Steps to Reproduce**
Follow query as above with incorrect geometry column name

**Expected behavior**
Standard error text with proper status code

**Comment by tomkralidis** [comment]
@krishnaglodha sure; can you provide steps to reproduce?

**Comment by krishnaglodha** [comment]
Hi @tomkralidis , to reproduce error, try 
`http://localhost:4501/collections/Popular%20places%20from%20database/items?f=json&filter=WITHIN(geom, POLYGON ((70.12683873493475 20.939807391818093, 70.12683873493475 16.187786570989473, 76.0788015865644 16.187786570989473, 76.0788015865644 20.939807391818093)))`

basically any incorrect geometry

**Comment by tomkralidis** [comment]
@krishnaglodha iirc this is a PostgreSQL backend?  cc @KoalaGeo for comments as well.

**Comment by krishnaglodha** [comment]
*Steps to reproduce* 

1. Create postgis connection to pygeoapi

<img width="1087" alt="Screenshot 2023-01-21 at 2 23 22 PM" src="https://user-images.githubusercontent.com/47075664/213859682-f3bf4be1-44ac-41fa-8de7-0fcf8b336c5a.png">

2. Create Query with incorrect geometry column 
<img width="1100" alt="Screenshot 2023-01-21 at 2 23 01 PM" src="https://user-images.githubusercontent.com/47075664/213859665-25b2e23f-b00f-4448-b560-7c70b5b23ad8.png">


*Expected response*
```
{
    "code": "InvalidParameterValue",
    "description": "Bad CQL string : CONTAINS(geometry,POINT(80 20))"
}
```

**Comment by KoalaGeo** [comment]
Unfortunately you have to query whatever the geometry column is called on the original PostGres source. 

@tomkralidis It's on our to do as a fix, so the geometry column name is always mapped to "geometry". Theres the same issue with ID

**Comment by krishnaglodha** [comment]
while checking queryables, it shows `geometry` as column name
<img width="489" alt="Screenshot 2023-01-22 at 9 03 32 AM" src="https://user-images.githubusercontent.com/47075664/213899317-a82fc365-e43e-4b63-b127-5d3e159e748f.png">

**Comment by KoalaGeo** [comment]
Is it called geometry in PostGres? Did the query work if you use the PostGres geometry column name?

**Comment by tomkralidis** [comment]
We already support a `geom_field` (and `id_field`) in the provider config (https://docs.pygeoapi.io/en/latest/data-publishing/ogcapi-features.html#postgresql).

Can the PostgreSQL provider detect if this is defined in config, and use it proper?  Else, derive geometry column name automatically from the `geometry_columns` table?

**Comment by KoalaGeo** [comment]
@tomkralidis that's what we want to do https://github.com/BritishGeologicalSurvey/pygeoapi/issues/19 have a sprint booked in for late Feb

**Comment by krishnaglodha** [comment]
@KoalaGeo can I participate in the sprint online? I would love to work on this issue.

**Comment by KoalaGeo** [comment]
@krishnaglodha sure we can sort something out, we'll be working against https://github.com/BritishGeologicalSurvey/pygeoapi then submit PR to geopython

**Comment by krishnaglodha** [comment]
@KoalaGeo great, you can keep me posted here or at me@krishnaglodha.com

**Comment by ricardogsilva** [comment]
FYI, support for the generic name `geometry` has now been merged in #1453 and there is also #1446 which should improve error visibility when merged. Not sure if those will be enough for closing this issue.

Can you please confirm if it can be closed @krishnaglodha ?
